infra/Pos/Slotting/Class.hs

```diff
@@ -22,13 +22,6 @@ class MonadSlotsData m => MonadSlots m where
     -- return until current slot is known.
     getCurrentSlotBlocking :: m SlotId
 
-    -- | This function tries to predict current slot as accurately as it can.
-    -- If 'getCurrentTime' returns unreliable time,
-    -- then function returns last known slot
-    -- If our slotting data into DB is outdated,
-    -- then function tries to extrapolate slot using last know slotting data
-    getCurrentSlotInaccurate :: m SlotId
```

Comment:
Why did you remove this function? Even if it wasn't used, it is still quite useful.
Even if we store all known slotting data, this function is still needed.
We don't need to extrapolate time when we want to know timestamp of the _passed_ slot. That's exactly the case for transactions from blocks. If we processed block, we know slotting data for it. That's why I say extrapolation isn't needed to compute transaction's timestamp.
However, if you didn't run your node for a long time, you don't know the current slot until you receive all blocks. Even if you store whole slotting data. But you may want to know the current slot at least approximately. So this function is useful in such cases.